### PR TITLE
FastCV-based HAL for OpenCV acceleration 2ndpost-1

### PR DESCRIPTION
**Pull request by quic-xuezha**
### Detailed description:

- Add parallel support for cv_hal_sobel
- Add cv_hal_gaussianBlurBinomial and parallel support.
- Add cv_hal_addWeighted8u and parallel support
- Add cv_hal_warpPerspective and parallel support

Requires binary from [opencv/opencv_3rdparty#90](https://github.com/opencv/opencv_3rdparty/pull/90)
Related patch to opencv_contrib: [opencv/opencv_contrib#3844](https://github.com/opencv/opencv_contrib/pull/3844)

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [ ] The PR is proposed to the proper branch
- [ ] There is a reference to the original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

**Comment by asmorkalov** [comment]
@quic-xuezha Thanks a lot for the contribution! Could you enable FastCV by default in the PR to check it with CI. We will revert the change just before merge.

**Comment by asmorkalov** [comment]
There is no FastCV links & hashes update in CMake. Old library will be used.

File: 3rdparty/fastcv/src/fastcv_hal_imgproc.cpp [hunk 1]
```diff
@@ -141,28 +221,88 @@ int fastcv_hal_sobel(
...
+        }
+
+        uchar* dptr = dst_data;
+        uchar* sptr = tmp.data() + start_height*dst_step + start_width;
+        for (int i = 0; i < height; ++i)
+        {
+            memcpy(dptr, sptr, width*sizeof(int16_t));
+            dptr += dst_step;
+            sptr += dst_step;
+        }
```
Comment (asmorkalov) [inline 3rdparty/fastcv/src/fastcv_hal_imgproc.cpp:296]: Can you get rid of memcopy?

File: 3rdparty/fastcv/src/fastcv_hal_imgproc.cpp [hunk 2]
```diff
@@ -316,3 +456,278 @@ int fastcv_hal_adaptiveThreshold(
...
+        std::vector<uchar> dst(dst_step * rangeHeight);
+
+        if (fcvFuncType == FCV_MAKETYPE(3,CV_8U))
+            fcvFilterGaussian3x3u8_v4(src, width, rangeHeight, src_step, dst.data(), dst_step, fcvBorder, 0);
+        else if (fcvFuncType == FCV_MAKETYPE(5,CV_8U))
+            fcvFilterGaussian5x5u8_v3(src, width, rangeHeight, src_step, dst.data(), dst_step, fcvBorder, 0);
+
+        uchar *dptr = dst_data + range.start * dst_step;
+        uchar *sptr = dst.data() + topLines * dst_step;
+        memcpy(dptr, sptr, (range.end - range.start) * dst_step);
```
Comment (asmorkalov) [inline 3rdparty/fastcv/src/fastcv_hal_imgproc.cpp:499]: The same question on copy.

File: 3rdparty/fastcv/src/fastcv_hal_imgproc.cpp (hunk 1 above)
Comment (asmorkalov) [reply 3rdparty/fastcv/src/fastcv_hal_imgproc.cpp:296]: OK, I got why you need extra buffer and memcpy. Please add comment why you did it into the code. it's not obvious for readers.

File: 3rdparty/fastcv/src/fastcv_hal_imgproc.cpp (hunk 2 above)
Comment (asmorkalov) [reply 3rdparty/fastcv/src/fastcv_hal_imgproc.cpp:499]: The same proposal.

File: 3rdparty/fastcv/src/fastcv_hal_imgproc.cpp (hunk 1 above)
Comment (quic-xuezha) [reply 3rdparty/fastcv/src/fastcv_hal_imgproc.cpp:296]: when the input is ROI, we need additional lines to on top and bottom to be border, for example ksize is 3, we need 1 more line at top and 1 more line at bottom, the input height for fastcv function will become range.end-range.start+2, we use a tmp var to store this result and use memcpy to copy it back to dst.

But now I have removed memcpy and use cv::Mat and copyTo function, will update the code later.

**Comment by asmorkalov** [comment]
Linux configuration build issue:
```
-- FASTCV: Downloading fastcv_linux_aarch64_2024_12_11.tgz from https://raw.githubusercontent.com/opencv/opencv_3rdparty/1eb84499ee9996616529fc48b0bab5631f64c2dc/fastcv/fastcv_linux_aarch64_2024_12_11.tgz
CMake Warning at cmake/OpenCVDownload.cmake:271 (message):
  FASTCV: Hash mismatch: a194679b934516dc2d377bb33e71ae76
Call Stack (most recent call first):
  3rdparty/fastcv/fastcv.cmake:29 (ocv_download)
  cmake/OpenCVFindLibsPerf.cmake:187 (download_fastcv)
  CMakeLists.txt:797 (include)


CMake Warning at 3rdparty/fastcv/fastcv.cmake:40 (message):
  FastCV: package download failed!
Call Stack (most recent call first):
  cmake/OpenCVFindLibsPerf.cmake:187 (download_fastcv)
  CMakeLists.txt:797 (include)
```

**Comment by quic-xuezha** [comment]
will check and update the hash

**Review by asmorkalov — APPROVED** [review]

**Comment by asmorkalov** [comment]
@quic-xuezha Thanks a lot for the contribution. I squashed the commits in opencv_3rdparty repo and created branch [fastcv/4.x_20241211](https://github.com/opencv/opencv_3rdparty/tree/fastcv/4.x_20241211) there. Please update links/hashes again and revert the change that enables FastCV by default. I'll merge the PR after it.

**Comment by quic-xuezha** [comment]
Updated the commit id and reverted the change

**Pull request merged**